src/Storages/MergeTree/MutatePlainMergeTreeTask.h

```diff
@@ -48,8 +48,8 @@ class MutatePlainMergeTreeTask : public IExecutableTask
     String getQueryId() const override { return getStorageID().getShortName() + "::" + merge_mutate_entry->future_part->name; }
 
 private:
-
     void prepare();
+    void finish();
```

Comment:
Do we need the same change for other tasks, like `ReplicatedMergeMutateTaskBase`? What's the difference between `finish` and `onCompleted`?

Reply:
Looks like we need a similar change for all tasks (`*PlainMergeTreeTask` is for ordinary MergeTree, there are other tasks for RMT and SMT), consider adding `finish` to the interface as well (with a comment about `onCompleted`)

Reply:
@tavplubix I checked it `MergeMutateSelectedEntry` + all the problems I found in there are used and related only to plain MergeTree...